### PR TITLE
Added Password Protection and biometric authentication by Encrypted Shared Preferences and biometrics in android

### LINKED ISSUE #115: Protection
Some users would like an option for the app to require a password or biometric input before opening.
I am not interested in this but a PR would be welcome of course.
Keep in mind that this feature will be [integrated into Google Play services](https://blog.google/products/android/android-theft-protection/) next year.

### PR DESCRIPTION
**Pull request by anuragkanojiya1**
## Related Issue
Closes: #115 

## Description
Implemented the password protection feature using encrypted shared preferences. This ensures that sensitive user data is securely stored in an offline mode and accessed only through proper authentication. Added logic to enable and disable password protection, and updated the settings screen to reflect the current state of the password protection feature.

## Type of Change
https://github.com/user-attachments/assets/3d63c8f8-298a-426f-9710-4e43fe14232b

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [x] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [x] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by CrazyMarvin** [comment]
This is awesome!
Thank you very much for your input, @anuragkanojiya1. 👍

Would it be possible to use biometrics instead of a password too?

May you please make the new strings translateable (android/app/src/main/res/values/strings.xml )?

Can you please have a look too, @mubeen1519? What do you think?

**Comment by mubeen1519** [comment]
sure will look into it but biometric feature are good approach for this app and convinent for user

**Comment by mubeen1519** [comment]
actually its looks not too good to use password and this design also if biometric option is available we dont need to do this stuff only prompt show at start for biometric and user dont need to write and remember the password

**Comment by anuragkanojiya1** [comment]
Okay, I will try to make a PR regarding biometric authentication soon.

**Comment by CrazyMarvin** [comment]
I think both options would be great. So the users can decide if they go without extra protection, password or biometrics.

**Comment by CrazyMarvin** [comment]
Thanks for giving a helping hand, @mubeen1519! ❤️

**Comment by anuragkanojiya1** [comment]
> I think both options would be great. So the users can decide if they go without extra protection, password or biometrics. 

Okay @CrazyMarvin

**Comment by anuragkanojiya1** [comment]
@CrazyMarvin I have implemented biometric auth and tested it on my physical device, it is working fine, you should check it out too.

**Comment by CrazyMarvin** [comment]
May you share a debug APK please, @anuragkanojiya1?
May you please review the code, @mubeen1519?

**Comment by anuragkanojiya1** [comment]
@CrazyMarvin Here's the debug apk.

[debug.zip](https://github.com/user-attachments/files/17350903/debug.zip)

**Comment by anuragkanojiya1** [comment]
@CrazyMarvin any update?

**Comment by mubeen1519** [comment]
> @CrazyMarvin Here's the debug apk.
> 
> [debug.zip](https://github.com/user-attachments/files/17350903/debug.zip)

the apk is not installing on my device android 10

File: android/app/src/main/java/rocks/poopjournal/fucksgiven/MainActivity.kt [hunk 1]
```diff
@@ -23,7 +25,7 @@ import rocks.poopjournal.fucksgiven.presentation.ui.utils.ThemeSetting
 import javax.inject.Inject
 
 @AndroidEntryPoint
-class MainActivity : ComponentActivity() {
+class MainActivity : AppCompatActivity() {
```
Comment (mubeen1519) [inline android/app/src/main/java/rocks/poopjournal/fucksgiven/MainActivity.kt:28]: Why it is changed into AppCompatActivity?

**Comment by anuragkanojiya1** [comment]
> > @CrazyMarvin Here's the debug apk.
> > 
> > [debug.zip](https://github.com/user-attachments/files/17350903/debug.zip)
> 
> the apk is not installing on my device android 10

It's working fine on my Android 10 device.

**Comment by CrazyMarvin** [comment]
I just merged your PR.

The UX needs a bit fine-tuning. Are you interested to do that next week?

Thanks!

**Comment by anuragkanojiya1** [comment]
> I just merged your PR.
> 
> The UX needs a bit fine-tuning. Are you interested to do that next week?
> 
> Thanks!

---

Sure @CrazyMarvin,
I’d love to connect with you on LinkedIn. You can find my profile here: [Anurag Kanojiya](https://linkedin.com/in/anurag-kanojiya-101312286). 

Additionally, if you enjoy my work and would like to support me, you can check out my GitHub Sponsors profile: [Sponsor My Work](https://github.com/sponsors/anuragkanojiya1). Your support would mean a lot to me!

Looking forward to contribute more!

Best regards,  
Anurag Kanojiya

---

**Comment by CrazyMarvin** [comment]
The designer updated the mockup.
Are you able to create a new PR that reflects it, @anuragkanojiya1?

**Comment by anuragkanojiya1** [comment]
> The designer updated the mockup. Are you able to create a new PR that reflects it, @anuragkanojiya1?

Sure @CrazyMarvin.

**Pull request merged**